### PR TITLE
ScrollFloat: allow a content context used to measure distance from bottom of content vs. scroll context.

### PR DESCRIPTION
**Pull request by mikesherov**

### CODE REVIEW

File: ux/scrollfloat.js [hunk 1]
```diff
@@ -100,39 +128,46 @@ define([
...
+    for (breakpoint in registry[contextId]) {
+      cb = registry[contextId][breakpoint];
       // search for the original function
       i = cb.original.indexOf(fn);
       if (~i) {
         cb.original.splice(i, 1);
         cb.wrapped.splice(i, 1);
         if (!cb.original.length) {
-          delete registry[context][breakpoint];
+          delete registry[contextId][breakpoint];
```
Comment (eerikson) [inline ux/scrollfloat.js]: I know you didn't really come up with the idea for `delete` but could this be upgraded to `splice` instead?

File: ux/scrollfloat.js [hunk 2]
```diff
@@ -100,39 +128,46 @@ define([
...
         }
       }
     }
 
-    if (!Object.keys(registry[context]).length) {
-      $context.off('scroll', scrollCache[context]);
-      delete registry[context];
+    if (!Object.keys(registry[contextId]).length) {
+      $context.off('scroll', scrollCache[contextId]);
+      delete registry[contextId];
```
Comment (eerikson) [inline ux/scrollfloat.js]: See above re: `delete` vs `splice`

File: ux/scrollfloat.js [hunk 3]
```diff
@@ -50,29 +75,32 @@ define([
...
-  function scrollfloat(breakpoint, callback, context) {
+  function scrollfloat(breakpoint, callback, context, contentContext) {
     if (typeof breakpoint === 'function') {
+      contentContext = context;
       context = callback;
       callback = breakpoint;
       breakpoint = 1;
     }
     context = context || 'window';
     breakpoint = Number(breakpoint).toString();
```
Comment (nemtsov) [inline ux/scrollfloat.js]: I know you didn't update this piece, but according to the comment, a breakpoint could be `50%`. `Number('50%').toString()` returns `"NaN"`.

**Comment by nemtsov** [comment]
I think it would be great to start extracting libraries such as this one into beff-independent libraries, until beff is left with nothing.

File: ux/scrollfloat.js (hunk 3 above)
Comment (mikesherov) [inline ux/scrollfloat.js]: yeah, I don't know why it does this... it should just be a Number to begin with... Don't feel like playing Jenga right now though.

**Comment by eerikson** [comment]
👍

**Comment by agaripian** [comment]
👍

File: ux/scrollfloat.js (hunk 1 above)
Comment (mikesherov) [inline ux/scrollfloat.js]: `registry[contextId][breakpoint]` is an object, not an array.

File: ux/scrollfloat.js (hunk 2 above)
Comment (mikesherov) [inline ux/scrollfloat.js]: `registry[contextId]` is an object, not an array.

File: ux/scrollfloat.js (hunk 1 above)
Comment (eerikson) [inline ux/scrollfloat.js]: Ah! Didn't realize this wasn't an index.

File: ux/scrollfloat.js (hunk 2 above)
Comment (eerikson) [inline ux/scrollfloat.js]: See above, worth considering IMO since this uses the `scroll` event.

File: ux/scrollfloat.js (hunk 1 above)
Comment (mikesherov) [inline ux/scrollfloat.js]: It's not only a safety thing, it's a correctness thing. `delete` will change the length of Object.keys, which is the goal.

Furthermore, this is not inside a scroll event, and is intended to be run once. Microbenchmarks on jsperf (BTW, I used to contribute code to that site!) is definitely a premature optimization here.

File: ux/scrollfloat.js (hunk 2 above)
Comment (mikesherov) [inline ux/scrollfloat.js]: See the above reply!

File: ux/scrollfloat.js (hunk 3 above)
Comment (mikesherov) [inline ux/scrollfloat.js]: addressed the comment, because `50%` would be totally broken.

File: ux/scrollfloat.js (hunk 1 above)
Comment (eerikson) [inline ux/scrollfloat.js]: Very cool and totally fair!

**Pull request merged**